From e888779d9c07e22a6aa58128d4777c138bddcaa8 Mon Sep 17 00:00:00 2001
From: Alexander Bauer <a.bauer@phytec.de>
Date: Wed, 4 Oct 2023 18:05:46 +0200
Subject: [PATCH] add_etml1010g3dra

Signed-off-by: Alexander Bauer <a.bauer@phytec.de>
---
 drivers/gpu/drm/panel/panel-simple.c | 29 ++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 3192411218fb..0dc8159a67e8 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -1776,6 +1776,32 @@ static const struct panel_desc edt_etml0700y5dha = {
 	.connector_type = DRM_MODE_CONNECTOR_LVDS,
 };
 
+static const struct drm_display_mode edt_etml1010g3dra_mode = {
+        .clock = 66500,
+        .hdisplay = 1280,
+        .hsync_start = 1280 + 12,
+        .hsync_end = 1280 + 12 + 20,
+        .htotal = 1280 + 12 + 20 + 56,
+        .vdisplay = 800,
+        .vsync_start = 800 + 1,
+        .vsync_end = 800 + 1 + 3,
+        .vtotal = 800 + 1 + 3 + 20,
+};
+
+static const struct panel_desc edt_etml1010g3dra = {
+        .modes = &edt_etml1010g3dra_mode,
+        .num_modes = 1,
+        .bpc = 8,
+        .size = {
+                .width = 216,
+                .height = 135,
+        },
+        .bus_format = MEDIA_BUS_FMT_RGB888_1X7X4_SPWG,
+        .bus_flags = DRM_BUS_FLAG_DE_HIGH,
+        .connector_type = DRM_MODE_CONNECTOR_LVDS,
+};
+
+
 static const struct drm_display_mode edt_etmv570g2dhu_mode = {
 	.clock = 25175,
 	.hdisplay = 640,
@@ -4114,6 +4140,9 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "edt,etml0700y5dha",
 		.data = &edt_etml0700y5dha,
+	}, {
+		.compatible = "edt,etml1010g3dra",
+		.data = &edt_etml1010g3dra,
 	}, {
 		.compatible = "edt,etmv570g2dhu",
 		.data = &edt_etmv570g2dhu,
-- 
2.25.1

